server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/MappingRoutesTest.java

```diff
@@ -0,0 +1,248 @@
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static org.hamcrest.CoreMatchers.is;
+import static org.mockito.Mockito.mock;
+
+import javax.mail.internet.AddressException;
+
+import net.javacrumbs.jsonunit.core.Option;
+import org.apache.james.core.Domain;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.User;
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.domainlist.memory.MemoryDomainList;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.rrt.memory.MemoryRecipientRewriteTable;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.filter.log.LogDetail;
+import io.restassured.http.ContentType;
+
+class MappingRoutesTest {
+
+    private WebAdminServer webAdminServer;
+    private MemoryRecipientRewriteTable recipientRewriteTable;
+
+    @BeforeEach
+    void beforeEach() throws DomainListException {
+        JsonTransformer jsonTransformer = new JsonTransformer();
+        recipientRewriteTable = new MemoryRecipientRewriteTable();
+        DNSService dnsService = mock(DNSService.class);
+        DomainList domainList = new MemoryDomainList(dnsService);
+        domainList.addDomain(Domain.of("domain.tld"));
+        domainList.addDomain(Domain.of("aliasdomain.tld"));
+        domainList.addDomain(Domain.of("gmail.com"));
+
+        recipientRewriteTable.setDomainList(domainList);
+
+        webAdminServer = WebAdminUtils.createWebAdminServer(new MappingRoutes(jsonTransformer, recipientRewriteTable))
+            .start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(MappingRoutes.BASE_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void getMappingsShouldReturnEmptyWhenNoMappings() {
+        when()
+            .get()
+        .then()
+            .contentType(ContentType.JSON)
+            .statusCode(HttpStatus.OK_200)
+            .body(is("{}"));
+    }
+
+    @Test
+    void getMappingsShouldReturnAliasMappings() throws RecipientRewriteTableException {
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("alias@domain.tld")),
+            "user@domain.tld");
+
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("alias@domain.tld")),
```

Comment:
Extract the mapping source to a common variable to highligh that it is the same.